src/state/account.ts

```diff
@@ -189,14 +217,40 @@ export const accountSlice = createSlice({
     setTradingRewards: (state, action: PayloadAction<TradingRewards>) => {
       state.tradingRewards = action.payload;
     },
-    addUncommittedOrderClientId: (state, action: PayloadAction<number>) => {
-      state.uncommittedOrderClientIds.push(action.payload);
+    submittedOrder: (
+      state,
+      action: PayloadAction<{ marketId: string; clientId: number; orderType: TradeTypes }>
+    ) => {
+      state.submittedOrders.push({
+        ...action.payload,
+        submissionStatus: OrderSubmissionStatuses.Submitted,
+      });
+      state.uncommittedOrderClientIds.push(action.payload.clientId);
     },
-    removeUncommittedOrderClientId: (state, action: PayloadAction<number>) => {
+    submittedOrderFailed: (
+      state,
+      action: PayloadAction<{ clientId: number; errorStringKey: StringKey }>
+    ) => {
+      state.submittedOrders = state.submittedOrders.map((order) =>
+        order.clientId === action.payload.clientId
+          ? {
+              ...order,
+              errorStringKey: action.payload.errorStringKey,
+            }
+          : order
+      );
       state.uncommittedOrderClientIds = state.uncommittedOrderClientIds.filter(
-        (id) => id !== action.payload
+        (id) => id !== action.payload.clientId
       );
     },
+    submittedOrderTimeout: (state, action: PayloadAction<number>) => {
+      if (state.uncommittedOrderClientIds.includes(action.payload)) {
+        submittedOrderFailed({
+          clientId: action.payload,
+          errorStringKey: STRING_KEYS.SOMETHING_WENT_WRONG as StringKey,
```

Comment:
is the typing for `StringKey` incorrect? seems weird that you need to cast when `StringKey` should be derived from `STRING_KEYS`.

Reply:
umm so `StringKey` is actually the key.. so "SOMETHING_WENT_WRONG" would be a StringKey, but `STRING_KEYS.SOMETHING_WENT_WRONG` would not because it's actually 'ERRORS.GENERAL.SOMETHING_WENT_WRONG'..

adding a type for the value `(typeof STRING_KEYS)[StringKey]` just type it as a `string` T_T

i'll change it back to string for now